Src/Couchbase.Linq.Tests/QueryGeneration/NestTests.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Couchbase.Core;
+using Couchbase.Linq.Extensions;
+using Couchbase.Linq.Tests.Documents;
+using Moq;
+using NUnit.Framework;
+
+namespace Couchbase.Linq.Tests.QueryGeneration
+{
+    [TestFixture]
+    public class NestTests : N1QLTestBase
+    {
+        [Test]
+        public void Test_Unnest_Simple()
+        {
+            var mockBucket = new Mock<IBucket>();
+            mockBucket.SetupGet(e => e.Name).Returns("default");
+
+            var query = from brewery in QueryFactory.Queryable<Brewery>(mockBucket.Object)
+                        from address in brewery.Address
+                        select new {name = brewery.Name, address};
+
+            const string expected = "SELECT brewery.name as name, address as address " +
+                "FROM default as brewery "+
+                "INNER UNNEST brewery.address as address";
+
+            var n1QlQuery = CreateN1QlQuery(mockBucket.Object, query.Expression);
+
+            Assert.AreEqual(expected, n1QlQuery);
+        }
+
+        [Test]
+        public void Test_Unnest_Sort()
+        {
+            var mockBucket = new Mock<IBucket>();
+            mockBucket.SetupGet(e => e.Name).Returns("default");
+
+            var query = from brewery in QueryFactory.Queryable<Brewery>(mockBucket.Object)
```

Comment:
@brantburnett -

TBH, I wasn't aware of the double from clauses. Thanks for sharing.

I agree, extension methods are the way to go with NEST until perhaps C# 6.0 and Roslyn, but I am not 100% about that. I also agree to leave the syntax the same for now, this can be handled in documentation.

Also, If you would like to discuss, my email is jeffry.morris at couchbase dot com. Send me an email and we can do a Skype chat.

Thanks!

-Jeff